unblob/handlers/archive/arc.py

```diff
@@ -50,22 +50,26 @@ def calculate_chunk(
         self, file: io.BufferedIOBase, start_offset: int
     ) -> Optional[ValidChunk]:
 
-        # we loop from header to header until we reach the end header
-        offset = start_offset
-        while True:
-            file.seek(offset)
-            try:
+        try:
+            # we loop from header to header until we reach the end header
+            offset = start_offset
+            while True:
+                file.seek(offset)
                 read_bytes = file.read(2)
-            except EOFError:
-                logger.warning("Potential ARC header is missing")
-                return
-
-            if read_bytes == END_HEADER:
-                offset += 2
-                break
-            file.seek(offset)
-            header = self.parse_header(file)
-            offset += len(header) + header.size
+
+                if read_bytes == END_HEADER:
+                    offset += 2
+                    break
+                file.seek(offset)
+                header = self.parse_header(file)
+                offset += len(header) + header.size
+
+        except (EOFError):
+            # OSError triggered when seeking past the EOF
```

Comment:
On a related note: what throws the `EOFError`? You should always be able to read from an open file. After some digging I found that cstruct can raise it. I'd adjust this comment accordingly
```suggestion
        except EOFError:
            # EOFError triggered when file ends during header parsing
```

Reply:
If you put the try: except: block around the self.parse_header call, you don't need the comment. EOFError can't be raised from other parts of this code.

Reply:
Got rid of the comment and placed the try/except block around `parse_header` :)